Add mock for protocol MockFirmwareVolumeBlock under MdePkg

Rasheed-Yusuf opened the pull request:
## Description

Add Mock for Firmware Volume Block Protocol in MdePkg

Signed-off-by: rasheed yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [x] Backport to release branch?

## How This Was Tested

Consumed in a GoogleTest

## Integration Instructions

N/A

Rasheed-Yusuf committed:
Add mock protocol under MdePkg.

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockFirmwareVolumeBlock.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

Rasheed-Yusuf commented:
@microsoft-github-policy-service agree [company="Microsoft"]

________________________________
From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Monday, November 11, 2024 11:47 PM
To: microsoft/mu_basecore ***@***.***>
Cc: Rasheed Yusuf (PHOENIX TECHNOLOGIES Inc) ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/mu_basecore] Add mock for protocol MockFirmwareVolumeBlock under MdePkg (PR #1216)


@Rasheed-Yusuf<https://github.com/Rasheed-Yusuf> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]


Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree


  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"


Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement (“Agreement”) is agreed to by the party signing below (“You”),
and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
“Code” means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
“Project” means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<http://www.opensource.org/>).
“Submit” is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”
“Submission” means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase “Submission
containing materials of a third party:” followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
guidelines concerning Submissions.
  4.  Your Employer. References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/mu_basecore/pull/1216#issuecomment-2469813698>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BDU6GFXPJQXLQVMESO54O6D2AGXBRAVCNFSM6AAAAABRTM7UGWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRZHAYTGNRZHA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Rasheed-Yusuf commented:
@microsoft-github-policy-service agree

________________________________
From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Tuesday, November 12, 2024 12:00 AM
To: microsoft/mu_basecore ***@***.***>
Cc: Rasheed Yusuf (PHOENIX TECHNOLOGIES Inc) ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/mu_basecore] Add mock for protocol MockFirmwareVolumeBlock under MdePkg (PR #1216)


@Rasheed-Yusuf<https://github.com/Rasheed-Yusuf> the command you issued was incorrect. Please try again.

Examples are:

@microsoft-github-policy-service agree


and

@microsoft-github-policy-service agree company="your company"


—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/mu_basecore/pull/1216#issuecomment-2469833803>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BDU6GFQELR3D3PSGDO6EWF32AGYS7AVCNFSM6AAAAABRTM7UGWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRZHAZTGOBQGM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:
```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

Rasheed-Yusuf committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h

Co-authored-by: Michael Kubacki <michael.kubacki@microsoft.com>
Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockFirmwareVolumeBlock.h 

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockFirmwareVolumeBlock.h with CI fixes

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockFirmwareVolumeBlock.h to modify format for CI build

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:
```suggestion
struct MockEfiFirmwareVolumeBlockProtocol {

```

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:
Based on the latest discussions in Edk2, the definition of MockEfiSmmSwDispatch2Protocol should be a #define. Please see the example below:

https://github.com/tianocore/edk2/pull/5950/files#diff-bce28afad342dcfbadd547d3d185dc2159e235b39e204990b683711c22dfc5e9

Rasheed-Yusuf committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h

Co-authored-by: Vivian Nowka-Keane <vnowkakeane@microsoft.com>
Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockFirmwareVolumeBlock.h

Review changes to use #define for MOCK_EFI_FIRMWARE_VOLUME_BLOCK_PROTOCOL_INSTANCE definition.

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockFirmwareVolumeBlock.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockFirmwareVolumeBlock.h

Fixes for CI Build errors

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>
